crates/volta-core/src/tool/node/resolve.rs

```diff
@@ -156,34 +156,28 @@ fn read_cached_opt(url: &str) -> Fallible<Option<RawNodeIndex>> {
         file: expiry_file.to_owned(),
     })?;
 
-    if let Some(date) = &expiry {
-        let expiry_date = Expires::decode(
-            &mut [date
-                .parse()
-                .with_context(|| ErrorKind::ParseNodeIndexExpiryError)?]
-            .iter(),
-        )
-        .with_context(|| ErrorKind::ParseNodeIndexExpiryError)?;
-
-        let current_date = Expires::from(SystemTime::now());
+    if !expiry
+        .map(|date| httpdate::parse_http_date(&date))
+        .transpose()
+        .with_context(|| ErrorKind::ParseNodeIndexExpiryError)?
+        .is_some_and(|expiry_date| SystemTime::now() < expiry_date)
+    {
+        return Ok(None);
```

Comment:
It's definitely possible I'm misreading, but I believe this might be inverting the condition. In the current code, when `SystemTime::now() < expiry_date`, then we read the cached file. However, in this one we're returning no cache with that same condition.

Reply:
I would think the behavior would be the same as the code before this change as the negation of this condition is evaluated.